comp-mng-server/src/main/java/novi/uni/compserver/services/RegistrationService.java

```diff
@@ -26,8 +35,32 @@ public ApiResponse isEmployeeCreated(String username) {
         }
 
         NoviEmployee newEmployee = new NoviEmployee(username, passwordEncoder.encode(tempPass));
+
+        Role employeeRole = roleRepository.findByRolename(RoleName.ROLE_EMPLOYEE)
+                .orElseThrow(() -> new RoleNotFoundException("Rolename EMPLOYEE kan niet gevonden worden"));
+
+        newEmployee.setRoleSet(Collections.singleton(employeeRole));
+
         noviEmployeeRepository.save(newEmployee);
 
         return new ApiResponse("Nieuwe Novimedewerker is aangemaakt", HttpStatus.OK);
     }
+
+    public boolean isAdminCreated(String username) {
```

Comment:
Zelfde commentaar als bij isEmployeeCreated().
Deze 2 functies zijn erg vergelijkbaar, ik zou ze samenvoegen tot 2 meer transparante functies:

findEmployee(String userName) -> simpele find
createEmployee(String userName, RoleName roleName) -> maak die gebruiker

Dan in de aanroepende code (kan mooier met optionals maar ter illustratie:

Employee employee = registrationService.findEmployee("xyz");
if( employee == null ) {
    employee = registrationService.createEmployee("xyz", RoleNames.ADMIN )
}

Reply:
Als ik het goed begrijp, wordt de optional al impliciet gebruikt omdat ik er een orElseThrow lambda achteraan gooi wanneer die null  is. Anders heb ik vind ik vet lelijke code met if checks met Optional.isPresent en binnen de if block logica + geen lelijke null checks^^